fit_multiband.py

```diff
@@ -0,0 +1,255 @@
+    ConfigClass = MultibandFitConfig
+    _DefaultName = "multibandFit"
+
+    def __init__(self, initInputs, **kwargs):
+        super().__init__(initInputs=initInputs, **kwargs)
+        self.makeSubtask("fit_multiband", schema=initInputs["cat_ref_schema"].schema)
+        self.cat_output_schema = afwTable.SourceCatalog(self.fit_multiband.schema)
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        input_objs = [inputRefs.cats_meas, inputRefs.coadds]
+        dataIds_cats, dataIds = [[dRef.dataId for dRef in obj] for obj in input_objs]
+        if not all((x == y for x, y in zip(dataIds, dataIds_cats))):
+            raise RuntimeError(f'Mismatched cats_meas, coadds dataIds {dataIds_cats} {dataIds}')
+        bands_data = [dataId['band'] for dataId in dataIds]
+        bandset_fit, bandset_read = self.config.get_band_sets()
+        bandset_needed = bandset_fit.union(bandset_read)
+        bandset_data = set(bands_data)
+        if bandset_data != bandset_needed:
+            raise RuntimeError(f'set(band={bands_data})={bandset_data} != bandset_needed={bandset_needed}'
+                               f'=(bands_fit={bandset_fit}).union(bands_read={bandset_read}))')
+        cat_ref, cats_meas, coadds = (inputs[x] for x in ('cat_ref', 'cats_meas', 'coadds'))
+        n_data = [len(x) for x in (bands_data, cats_meas, coadds)]
+        if not all(x == n_data[0] for x in n_data[1:]):
+            raise RuntimeError(f'bands, cats_meas, coadds, dataIds lens={n_data} not matched')
+        catexps = [
+            CatalogExposure(band=band, catalog=cat, exposure=exp, dataId=dataId, metadata={},
+                            calib=exp.getPhotoCalib() if hasattr(exp, 'getPhotoCalib') else None)
+            for band, cat, exp, dataId in zip(bands_data, cats_meas, coadds, dataIds)
+        ]
+        outputs = self.run(catexps=catexps, cat_ref=cat_ref)
+        butlerQC.put(outputs, outputRefs)
+
+    def run(self, catexps: List[CatalogExposure], cat_ref: afwTable.SourceCatalog) -> pipeBase.Struct:
+        """ Fit sources from a reference catalog using data from multiple exposures in the same region (patch).
+
+        Parameters
+        ----------
+        catexps : `typing.List [CatalogExposure]`
+            A list of catalog-exposure pairs in a given band. Subtasks may require there to only be one catexp per band,
+            and/or for the catalogs to contain HeavyFootprints with deblended images.
+        cat_ref : `lsst.afw.table.SourceCatalog`
+            A reference source catalog to fit. Subtasks may be configured fit only a subset of these sources.
+
+        Returns
+        -------
+        retStruct : `lsst.pipe.base.Struct`
+            A struct with a cat_output attribute containing the output measurement catalog.
+        """
+        cat_output = self.fit_multiband.run(catexps, cat_ref).output
+        # It would be best to validate this earlier but I'm not sure how
```

Comment:
Not much you can do; this is something subclasses are expected to just be responsible for.

But you might want to move this check to `runQuantum`, and do it _after_ the `put`, so if there is a problem, the user ends up with files they can go look at to see how they're wrong.  It'd be nice to also have it here in `run`, but in practice I think it'll be a rare enough problem that making it easy to debug when it happens is more important.